# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1267770679 -3600
#      Fri Mar 05 07:31:19 2010 +0100
# Node ID bc475cd491477202bf248d029ceb589db0dfcf30
# Parent  7df8f8194c9650df9f283d756dde975ec2a31e8b
avoid div by zero in fzero

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-03-05  Jaroslav Hajek  <highegg@gmail.com>
+
+	* optimization/fzero.m: Avoid division by zero.
+
 2010-03-04  Jaroslav Hajek  <highegg@gmail.com>
 
 	* optimization/fminbnd.m: Fix tests.
 
 2010-03-04  Jaroslav Hajek  <highegg@gmail.com>
 
 	* optimization/fminbnd.m: Fix termination tolerances.
 
diff --git a/scripts/optimization/fzero.m b/scripts/optimization/fzero.m
--- a/scripts/optimization/fzero.m
+++ b/scripts/optimization/fzero.m
@@ -282,17 +282,17 @@ function [x, fval, info, output] = fzero
     endif
     if (itype == 2)
       mba = mu * (b - a);
     endif
   endwhile
 
   ## Check solution for a singularity by examining slope
   if (info == 1)
-    if ((b - a) != 0 && abs ((fb - fa)/(b - a) / slope0) > max (1e6, 0.5/tolx))
+    if ((b - a) != 0 && abs ((fb - fa)/(b - a) / slope0) > max (1e6, 0.5/(eps+tolx)))
       info = -5;
     endif
   endif
 
   output.iterations = niter;
   output.funcCount = nfev;
   output.bracket = [a, b];
   output.bracketf = [fa, fb];
